File: src/game/entity/bomb/ConfettiExplosion.java
package game.entity.bomb;

import game.entity.models.Coordinates;
import game.entity.models.Direction;
import game.entity.models.Entity;
import game.entity.models.Explosive;
import game.utils.Paths;

public class ConfettiExplosion extends AbstractExplosion {
    public ConfettiExplosion(
            Entity owner,
            Coordinates coordinates,
            Direction direction,
            Explosive explosive) {
        super(owner, coordinates, direction, explosive);
    }

    public ConfettiExplosion(
            Entity owner,
            Coordinates coordinates,
            Direction direction,
            int distanceFromBomb,
            Explosive explosive) {
        super(owner, coordinates, direction, distanceFromBomb, explosive);
    }

    public ConfettiExplosion(
            Entity owner,
            Coordinates coordinates,
            Direction direction,
            int distanceFromExplosive,
            Explosive explosive,
            boolean canExpand) {
        super(owner, coordinates, direction, distanceFromExplosive, explosive, canExpand);
    }

    @Override
    protected Class<? extends AbstractExplosion> getExplosionClass() {
        return getClass();
    }

    @Override
    protected String getBasePath() {
        return String.format("%s/clown/clown_explosion/clown_explosion", Paths.getEnemiesFolder());
    }
}

File: src/game/entity/bomb/ConfettiExplosion.kt
package game.entity.bomb

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.entity.models.Direction
import game.entity.models.Entity
import game.entity.models.Explosive
import game.utils.Paths

class ConfettiExplosion : AbstractExplosion {
    constructor(
            owner: Entity,
            coordinates: Coordinates,
            direction: Direction,
            explosive: Explosive) : super(owner, coordinates, direction, explosive)

    constructor(
            owner: Entity,
            coordinates: Coordinates,
            direction: Direction,
            distanceFromBomb: Int,
            explosive: Explosive) : super(owner, coordinates, direction, distanceFromBomb, explosive)

    constructor(
            owner: Entity,
            coordinates: Coordinates,
            direction: Direction,
            distanceFromExplosive: Int,
            explosive: Explosive,
            canExpand: Boolean) : super(owner, coordinates, direction, distanceFromExplosive, explosive, canExpand)

    override val explosionClass: Class<out AbstractExplosion>
        get() = javaClass

    override fun getType(): EntityTypes = EntityTypes.ConfettiExplosion

    override fun getEntitiesAssetsPath(): String = "${Paths.enemiesFolder}/clown/clown_explosion/clown_explosion"
}
